File: store/src/main/java/com/nytimes/android/external/store3/util/NoKeyParser.java
package com.nytimes.android.external.store3.util;

import com.nytimes.android.external.store3.base.Parser;

import javax.annotation.Nonnull;

import io.reactivex.annotations.NonNull;

public class NoKeyParser<Key, Raw, Parsed> implements KeyParser<Key, Raw, Parsed> {
    private final Parser<Raw, Parsed> parser;

    public NoKeyParser(@Nonnull Parser<Raw, Parsed> parser) {
        this.parser = parser;
    }

    @Override
    public Parsed apply(@NonNull Key key, @NonNull Raw raw) throws ParserException {
        return parser.apply(raw);
    }
}

File: store/src/main/java/com/nytimes/android/external/store3/util/NoKeyParser.kt
package com.nytimes.android.external.store3.util

import com.nytimes.android.external.store3.base.Parser

class NoKeyParser<in Key, in Raw, out Parsed>(private val parser: Parser<Raw, out Parsed>) : KeyParser<Key, Raw, Parsed> {

    @Throws(ParserException::class)
    override suspend fun apply(key: Key, raw: Raw): Parsed {
        return parser.apply(raw)
    }
}
